cpp/include/ucxx/constructors.h

```diff
@@ -74,6 +75,16 @@ std::shared_ptr<RequestStream> createRequestStream(std::shared_ptr<Endpoint> end
                                                    size_t length,
                                                    const bool enablePythonFuture);
 
+std::shared_ptr<RequestMem> createRequestMem(std::shared_ptr<Endpoint> endpoint,
+                                             bool send,
+                                             void* buffer,
+                                             size_t length,
+                                             uint64_t remote_addr,
+                                             ucp_rkey_h rkey,
```

Comment:
As discussed offline, the changes were made on top of an older branch before https://github.com/rapidsai/ucxx/pull/121 was merged (on December 12), which reorganizes how data is passed to constructors and thus prevents adding several different arguments to them as is done [here](https://github.com/rapidsai/ucxx/pull/166/files#diff-99229522298ec9b3fb9e931133be93d2189a06e0cea3a180df26c5b0433556e4R54-R58), we now instead have [“data classes”](https://github.com/rapidsai/ucxx/blob/branch-0.36/cpp/include/ucxx/request_data.h) that include all operation-specific arguments and makes the API cleaner and less error-prone (i.e., with the user accidentally swapping arguments), as you can see in the [newest constructors](https://github.com/rapidsai/ucxx/blob/branch-0.36/cpp/include/ucxx/request_tag.h#L83). We will want to have this updated to the new style.